### PR TITLE
Added support for working with private repos

### PR DESCRIPTION
**Pull request by marctc**
With this changes we can work with mercurial private repos.

### CODE REVIEW

**Comment by pydanny** [comment]
Sorry for the very, very slow response. Would you mind rebasing from master so test coverage can be checked on all systems?

File: docs/usage.rst [hunk 1]
```diff
@@ -52,6 +52,15 @@ And if you want to specify a branch you can do that with::
 
     $ cookiecutter https://github.com/audreyr/cookiecutter-pypackage.git --checkout develop
 
+Works with private repos
+------------------------
+
+If you want to work repos that are not hosted in github or bitbucket you can indicate explicitly the
```
Comment (msabramo) [inline docs/usage.rst]: "work with repos"

**Comment by msabramo** [comment]
Looks useful. Can you rebase to resolve merge conflicts?

**Comment by pydanny** [comment]
Rebased and merged in! Grats @marctc!

**Comment by marctc** [comment]
thanks @pydanny!!

**Pull request closed without merging**